src/main.c

```diff
@@ -109,6 +109,7 @@ int     nlch = '\n';
 
 bool    tablesext, tablesverify, gentables;
 char   *tablesfilename=0,*tablesname=0;
+bool    is_tablesfilename_alloc;
```

Comment:
I've said before. Having a additional flag to track whether `tablesfilename` has been allocated doesn't make sense.
If `tablesfilename` will never point to a static or stack buffer, `tablesfilename == NULL` shall be sufficient to check that.

Also, [strdup()](http://pubs.opengroup.org/onlinepubs/9699919799/functions/strdup.html) **duplicates** the buffer the pointer point to, and the new buffer can be safely called free() against.
Please, have some common POSIX knowledge.

Your fix of this is incorrect.